replication/row_event.go

```diff
@@ -1182,6 +1217,29 @@ func decodeString(data []byte, length int) (v string, n int) {
 	return
 }
 
+func decodeStringLatin1(data []byte, length int) (v string, n int) {
+	// Define the Latin1 decoder
+	decoder := charmap.ISO8859_1.NewDecoder()
```

Comment:
Note that `latin1` in MySQL is `cp1252` and not `ISO 8859-1`:

"MySQL's latin1 is the same as the Windows cp1252 character set. This means it is the same as the official ISO 8859-1 or IANA (Internet Assigned Numbers Authority) latin1, except that IANA latin1 treats the code points between 0x80 and 0x9f as “undefined,” whereas cp1252, and therefore MySQL's latin1, assign characters for those positions. For example, 0x80 is the Euro sign. For the “undefined” entries in cp1252, MySQL translates 0x81 to Unicode 0x0081, 0x8d to 0x008d, 0x8f to 0x008f, 0x90 to 0x0090, and 0x9d to 0x009d. "

source: https://dev.mysql.com/doc/refman/8.4/en/charset-we-sets.html


```suggestion
	decoder := charmap.Windows1252.NewDecoder()
```